be/log.c

```diff
@@ -934,6 +940,17 @@ static int be_log_header_write(struct m0_be_log            *log,
 	return M0_RC(rc);
 }
 
+M0_INTERNAL int be_log_header_inc_generation(struct m0_be_log            *log,
+					     struct m0_be_fmt_log_header *log_hdr)
+{
+	M0_ENTRY("be_log_header_inc_generation=%"PRId64, log_hdr->flh_serial_num);
+	m0_be_log_header__set(log_hdr, log_hdr->flh_discarded, log_hdr->flh_group_lsn,
+			      log_hdr->flh_group_size, log_hdr->flh_serial_num + 1);
+	int rc = be_log_header_write(log, log_hdr);
+	M0_LEAVE("be_log_header_inc_generation=%"PRId64, log_hdr->flh_serial_num);
```

Comment:
Is this function protected by some mutex lock?

If not, concurrent calling of this function will cause race-conditions, and result in wrong serial number.

Reply:
This function is called from m0_be_engine_start with be_engine_lock(en) held.

Reply:
Please add assertion of lock at the beginning of this function.

Reply:
Why do we need the second parameter?
It is embedded as "struct m0_be_log::lg_header".
So, we can get it from the first param.

Reply:
So, this function only increases the serial number, but also synchronized it onto persistent storage.
This needs to be added into comments.

Reply:
Also we don't call it a "generation" number, but a serial number.